src/Streamly/Internal/Data/Stream/IsStream/Transform.hs

```diff
@@ -1024,7 +1034,7 @@ dropWhileM p m = fromStreamD $ D.dropWhileM p $ toStreamD m
 -- /Internal/
 --
 {-# INLINE dropByTime #-}
-dropByTime ::(MonadIO m, IsStream t, TimeUnit64 d) => d -> t m a -> t m a
+dropByTime ::(MonadAsync m, IsStream t, TimeUnit64 d) => d -> t m a -> t m a
```

Comment:
Behavior Change: same as in takeByTime.

We got the perf improvement due to the behavior change. Check the performance at lower timer granularity.